src/Chronos.php

```diff
@@ -182,6 +196,112 @@ public static function hasTestNow(): bool
         return static::$testNow !== null;
     }
 
+    /**
+     * Sets the time.
+     *
+     * @param int $hours Hours of the time
+     * @param int $minutes Minutes of the time
+     * @param int $seconds Seconds of the time
+     * @param int $microseconds Microseconds of the time
+     * @return static
+     */
+    public function setTime(int $hours, int $minutes, int $seconds = 0, int $microseconds = 0): static
+    {
+        $new = clone $this;
+        $new->native = $new->native->setTime($hours, $minutes, $seconds, $microseconds);
+
+        return $new;
+    }
+
+    /**
+     * Creates a new instance with date modified according to DateTimeImmutable::modifier().
+     *
+     * @param string $modifier Date modifier
+     * @return static
+     * @throws \InvalidArgumentException
+     */
+    public function modify(string $modifier): static
+    {
+        $native = $this->native->modify($modifier);
+        if ($native === false) {
+            throw new InvalidArgumentException('Unable to modify date using: ' . $modifier);
+        }
+
+        $new = clone $this;
+        $new->native = $native;
+
+        return $new;
+    }
+
+    /**
+     * Returns the difference between this instance and target.
+     *
+     * @param \Cake\Chronos\Chronos|\DateTimeInterface $target Target instance
+     * @param bool $absolute Whether the interval is forced to be positive
+     * @return \DateInterval
+     */
+    public function diff(Chronos|DateTimeInterface $target, bool $absolute = false): DateInterval
+    {
+        $target = $target instanceof DateTimeInterface ? $target : $target->native;
+
+        return $this->native->diff($target, $absolute);
+    }
+
+    /**
+     * Returns formatted date string according to DateTimeImmutable::format().
+     *
+     * @param string $format String format
+     * @return string
+     */
+    public function format(string $format): string
+    {
+        return $this->native->format($format);
+    }
+
+    /**
+     * Returns the timezone offset.
+     *
+     * @return int
+     */
+    public function getOffset(): int
+    {
+        return $this->native->getOffset();
+    }
+
+    /**
+     * Sets the date and time based on a Unix timestamp.
+     *
+     * @param int $timestamp Unix timestamp representing the date
+     * @return static
+     */
+    public function setTimestamp(int $timestamp): static
+    {
+        $new = clone $this;
+        $new->native = $new->native->setTimestamp($timestamp);
+
+        return $new;
+    }
+
+    /**
+     * Gets the Unix timestamp for this instance.
+     *
+     * @return int
+     */
+    public function getTimestamp(): int
+    {
+        return $this->native->getTimestamp();
+    }
+
+    /**
+     * Return time zone set for this instance.
+     *
+     * @return \DateTimeZone|false
+     */
+    public function getTimezone(): DateTimeZone|false
+    {
+        return $this->native->getTimezone();
```

Comment:
Are you aware in what case this would return `false`? Manual just says "...false in case of failure".

Reply:
I could not figure it out. If you look at the PHP source, it is false if some internal timelib flag "is local time" is set. I have no idea how that happens.

Reply:
I would change the return type for our method to just `DateTimeZone` and throw and exception if the native class returns false.

Reply:
Being more strict sounds good to me too.